From b632df5e663ab467176557501f3f13e23bb27a8e Mon Sep 17 00:00:00 2001
From: mbah37 <mbah37@student.cscc.edu>
Date: Wed, 12 Nov 2025 10:44:52 -0500
Subject: [PATCH 05/17] 2nd tutorial. created a settings file/class and changed
 the background to be an image scaled to the screen. (planning on changed
 background soon)

---
 alien_invasion.py | 17 +++++++++++++----
 settings.py       | 11 +++++++++++
 2 files changed, 24 insertions(+), 4 deletions(-)
 create mode 100644 settings.py

diff --git a/alien_invasion.py b/alien_invasion.py
index 9972627..498800f 100644
--- a/alien_invasion.py
+++ b/alien_invasion.py
@@ -1,16 +1,24 @@
 import sys
 import pygame
-
+from settings import Settings
 
 class AlienInvasion:
 
     def __init__(self):
         pygame.init()
+        self.settings = Settings()
+
+        self.screen = pygame.display.set_mode(
+            (self.settings.screen_width, self.settings.screen_height))
+        
+        pygame.display.set_caption(self.settings.name)
 
-        self.screen = pygame.display.set_mode((1200, 800))
-        pygame.display.set_caption("Alien Invasion")
+        self.bg = pygame.image.load(self.settings.bg_file)
+        self.bg = pygame.transform.scale(self.bg,
+             (self.settings.screen_width, self.settings.screen_height))
 
         self.running = True
+        self.clock = pygame.time.Clock()
     
     def run_game(self):
         while self.running:
@@ -19,8 +27,9 @@ class AlienInvasion:
                     self.running = False
                     pygame.quit()
                     sys.exit()
-            
+            self.screen.blit(self.bg, (0, 0))
             pygame.display.flip()
+            self.clock.tick(self.settings.FPS)
 
 if __name__ == '__main__':
     ai = AlienInvasion()
diff --git a/settings.py b/settings.py
new file mode 100644
index 0000000..a4a9828
--- /dev/null
+++ b/settings.py
@@ -0,0 +1,11 @@
+from pathlib import Path
+class Settings:
+
+    def __init__(self):
+        self.name : str = 'White Walker Invasion'
+        self.screen_width : int = 1200
+        self.screen_height : int = 800
+        self.FPS : int = 60
+        self.bg_file : Path = Path.cwd() / 'Assets' / 'images' / 'Starbasesnow.png'
+
+
-- 
2.51.0.windows.1

